cmd/kubeadm/app/util/version_test.go

```diff
@@ -525,3 +528,77 @@ func TestValidateStableVersion(t *testing.T) {
 func errorFetcher(url string, timeout time.Duration) (string, error) {
 	return "should not make internet calls", errors.Errorf("should not make internet calls, tried to request url: %s", url)
 }
+
+func TestFetchFromURL(t *testing.T) {
+	tests := []struct {
+		name     string
+		url      string
+		expected string
+		timeout  time.Duration
+		code     int
+		body     string
+	}{
+		{
+			name:     "normal success",
+			url:      "/normal",
+			expected: "normal response",
+			code:     http.StatusOK,
+			body:     "normal response",
+		},
+		{
+			name:     "HTTP error status",
+			url:      "/error",
+			expected: "",
+			code:     http.StatusBadRequest,
+			body:     "bad request",
+		},
+		{
+			name:     "Request timeout",
+			url:      "/timeout",
+			expected: "",
+			timeout:  time.Millisecond * 50,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			klog.V(2).Infof("Running test case: %s", tt.name)
+
+			handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				if tt.code != 0 {
+					w.WriteHeader(tt.code)
+				}
+				if tt.body != "" {
+					if _, err := w.Write([]byte(tt.body)); err != nil {
+						t.Error("Write body failed.")
+					}
+				}
+				if tt.timeout == time.Millisecond*50 {
+					time.Sleep(time.Millisecond * 200)
+					w.WriteHeader(http.StatusOK)
+					if _, err := w.Write([]byte("Delayed response")); err != nil {
+						t.Error("Write body failed.")
+					}
+				}
+			})
+
+			ts := httptest.NewServer(handler)
+			defer ts.Close()
+
+			url := ts.URL + tt.url
+
+			result, err := fetchFromURL(url, tt.timeout)
+
+			if tt.expected != "" {
+				if err != nil {
+					t.Errorf("Expected no error but got: %v", err)
+				}
+				if result != tt.expected {
+					t.Errorf("Expected '%s', but got '%s'", tt.expected, result)
+				}
+			} else if tt.expected != "" && err == nil {
```

Comment:
If `tt.expected != ""` the else if condition will never be executed

Reply:
Adjusted, thanks